projects/lights_control/src/lights_can.c

```diff
@@ -0,0 +1,109 @@
+#include "can.h"
+#include "can_hw.h"
+#include "can_msg_defs.h"
+
+#include "can_unpack_impl.h"
+#include "event_queue.h"
+#include "status.h"
+
+#include "lights_can.h"
+#include "lights_events.h"
+#include "lights_gpio.h"
+
+#define SOMEPORT 0
+#define SOMEPIN 0
+
+#define CASE_ACTION(action)            \
+  case ACTION_##action:                \
+    event_raise(EVENT_##action, data); \
+    break;
+
+#define CAN_RX_ADDR \
+  { 0, 11 }
+
+#define CAN_TX_ADDR \
+  { 0, 12 }
+
+// TODO(ELEC-372): remove these and get the device ID from elsewhere
+#define CAN_DEVICE_ID_FRONT 0
+#define CAN_DEVICE_ID_REAR 0
+
+static BoardType s_boardtype;
+
+static CANStorage s_can_storage;
+static CANRxHandler s_rx_handlers[CAN_NUM_RX_HANDLERS];
+
+static StatusCode prv_rx_handler(const CANMessage *msg, void *context, CANAckStatus *ack_reply);
+
+// TODO(ELEC-372):
```

Comment:
Is this `TODO` still valid?

Reply:
yeah where should I look for the RX/TX addresses? @tituschow  said to use the enums in `codegen-tooling` but I didn't find anything.

Reply:
The `enums` in `codegen-tooling` are used for message IDs and device IDs.

The `rx_event` and `tx_event` are the event IDs for events that get raised when a TX or RX event occurs (see [`can.c:172`](https://github.com/uw-midsun/firmware/blob/master/libraries/ms-common/src/can.c#L172) for RX and [`can.c:125`](https://github.com/uw-midsun/firmware/blob/master/libraries/ms-common/src/can.c#L125) for TX), which are whatever you define in your `lights_events.h` file.

Reply:
yeah I already have `LIGHTS_EVENTS_CAN_RX` and `LIGHTS_EVENTS_CAN_TX` defined, I'm looking for `CAN_TX_ADDR`, `CAN_RX_ADDR`, and the `bitrate`.

Reply:
oh my b, didn't realise you were talking about those

`CAN_TX_ADDR` and `CAN_RX_ADDR` are the GPIO port/pin combination, which can be found by looking on the [Controller Board schematic](https://github.com/uw-midsun/hardware/blob/master/MSXII_ControllerBoard/Project%20Outputs%20for%20Controller-Board/Controller_Board.pdf).

Reply:
Yeah, in this context where we have CAN device addresses, calling it `CAN_TX_ADDR` is a little confusing.

Reply:
changed it to `LIGHTS_CAN_TX_ADDR`